keyboards/gmmk/pro/config.h

```diff
@@ -39,27 +41,64 @@
 #define BOOTMAGIC_LITE_COLUMN 3
 
 #define TAP_CODE_DELAY 10
-#define ENCODERS_PAD_A { C15 }
-#define ENCODERS_PAD_B { C14 }
+#define ENCODERS_PAD_A \
+    { C15 }
+#define ENCODERS_PAD_B \
+    { C14 }
 
 /* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
 #define LOCKING_SUPPORT_ENABLE
 /* Locking resynchronize hack */
 #define LOCKING_RESYNC_ENABLE
 
-/* SPI Config for LED Driver */
 #define SPI_DRIVER SPID1
 #define SPI_SCK_PIN A5
-#define SPI_MOSI_PIN A6
-#define SPI_MISO_PIN A7
+#define SPI_SCK_PAL_MODE 5
+#define SPI_MISO_PIN A6
+#define SPI_MISO_PAL_MODE 5
+#define SPI_MOSI_PIN A7
+#define SPI_MOSI_PAL_MODE 5
 
-#define DRIVER_1_CS B13
-#define DRIVER_2_CS B14
-#define DRIVER_1_EN C13
-#define DRIVER_2_EN C13
+#define SPI_SS_DRIVER_1_PIN B13
+#define SPI_SS_DRIVER_2_PIN B14
+#define ENABLE_DRIVERS_PIN C13
+
+#define SW_LINES_ENABLE_DRIVER_1 SW_LINES_ENABLED_1_TO_12
+#define SW_LINES_ENABLE_DRIVER_2 SW_LINES_ENABLED_1_TO_12
 
 #define DRIVER_COUNT 2
 #define DRIVER_1_LED_TOTAL 66
 #define DRIVER_2_LED_TOTAL 32
 #define DRIVER_LED_TOTAL (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+#define RGB_MATRIX_CENTER \
+    { 112, 32 }
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 175
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_KEYPRESSES
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#    define ENABLE_RGB_MATRIX_ALPHAS_MODS
+#    define ENABLE_RGB_MATRIX_BAND_SAT
+#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
+#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
+#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
+#    define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
+#    define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
+#    define ENABLE_RGB_MATRIX_RAINBOW_BEACON
+#    define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
+#    define ENABLE_RGB_MATRIX_RAINDROPS
+#    define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#    define ENABLE_RGB_MATRIX_DIGITAL_RAIN
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
 
+#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_UP_DOWN
+#    define RGB_MATRIX_STARTUP_SAT 255
+#    define RGB_MATRIX_STARTUP_VAL 192
+#    define RGB_MATRIX_STARTUP_SPD 30
```

Comment:
Is there a reason these need to be enabled globally?
Wouldn't it make more sense to just let users enable whatever they need in their own keymaps.

Reply:
Also, the "enable" defines don't do anything. They're enabled by default, unless explicitly disabled. 

As for the startup values,  I think that pushing these to a `post_config.h` file, and having it check first would be best. Eg:
```c
#ifndef RGB_MATRIX_STARTUP_MODE
#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_UP_DOWN
#endif
#ifndef RGB_MATRIX_STARTUP_SAT
#    define RGB_MATRIX_STARTUP_SAT 255
#endif
```